gap/ReducibleMatrixGroups.gi

```diff
@@ -251,37 +262,112 @@ function(d, q, k)
                 generatorOfSUSubspace{[kHalf + 2..k]}{[1..kHalf]};
             generator{[1..kHalf]}{[d - kHalf + 1..d]} := 
                 generatorOfSUSubspace{[1..kHalf]}{[kHalf + 2..k]};
-            generator{[dHalf + 1]}{[1..kHalf]} := 
-                generatorOfSUSubspace{[kHalf + 1]}{[1..kHalf]};
-            generator{[dHalf + 1]}{[dHalf + 1]} := 
-                generatorOfSUSubspace{[kHalf + 1]}{[kHalf + 1]};
-            generator{[dHalf + 1]}{d - kHalf + 1..d]} := 
-                generatorOfSUSubspace{[kHalf + 1]}{[kHalf + 2..k]};
-            generator{[1..kHalf]}{[dHalf + 1]} := 
+            # use w_1 = alpha * e_{dHalf} + f_{dHalf} for the following lines
+            generator{[1..kHalf]}{[dHalf]} := 
+                alpha * generatorOfSUSubspace{[1..kHalf]}{[kHalf + 1]};
+            generator{[1..kHalf]}{[dHalf + 1]} :=
                 generatorOfSUSubspace{[1..kHalf]}{[kHalf + 1]};
-            generator{[d - kHalf + 1..d]}{[dHalf + 1]} := 
+            generator{[d - kHalf + 1..d]}{[dHalf]} :=
+                alpha * generatorOfSUSubspace{[kHalf + 2..k]}{[kHalf + 1]};
+            generator{[d - kHalf + 1..d]}{[dHalf]} :=
                 generatorOfSUSubspace{[kHalf + 2..k]}{[kHalf + 1]};
+            # now use e_{dHalf} = w_1 + beta ^ q * w_2
+            #         f_{dHalf} = alpha ^ q * w_1 - alpha * beta ^ q * w_2
+            generator{[dHalf]}{[1..kHalf]} :=
+                generatorOfSUSubspace{[kHalf + 1]}{[1..kHalf]}; 
+            generator{[dHalf + 1]}{[1..kHalf]} :=
+                alpha ^ q * generatorOfSUSubspace{[kHalf + 1]}{[1..kHalf]};
+            generator{[dHalf]}{[d - kHalf + 1..d]} :=
+                generatorOfSUSubspace{[kHalf + 1]}{[kHalf + 2..k]};
+            generator{[dHalf + 1]}{[d - kHalf + 1..d]} :=
+                alpha ^ q * generatorOfSUSubspace{[kHalf + 1]}{[kHalf + 2..k]};
```

Comment:
The code in this function is really difficult to read and thus also difficult to verify that it is correct. I have several suggestions to improve on that.

First off, I think long "speaking" names are not a boon here, but rather a malus: if you rename `generator` to `g` and `generatorOfSUSubspace` to `h`, for example, this particular section can be rewritten as:
```suggestion
            g{[dHalf]}{[1..kHalf]}             := h{[kHalf + 1]}{[1..kHalf]}; 
            g{[dHalf + 1]}{[1..kHalf]}         := h{[kHalf + 1]}{[1..kHalf]} * alpha ^ q;
            g{[dHalf]}{[d - kHalf + 1..d]}     := h{[kHalf + 1]}{[kHalf + 2..k]};
            g{[dHalf + 1]}{[d - kHalf + 1..d]} := h{[kHalf + 1]}{[kHalf + 2..k]} * alpha ^ q;
```
which is *much* easier (for me, at least) to take in "at a glance".

(In addition, I would recommend avoiding extremely similar and at the same time long names like `generatorOfSUSubspace` and `generatorsOfSUSubspace`, as it is very easy to confuse them with each other -- the single letter difference is easily lost in the big letter sea).

Reply:
@maxhauck ping?

Reply:
Well, I realised that there was no need to have the variable `generatorsOfSUSubspace` in the first place so it does not appear anymore in the current version of this code snippet, which neatly solves the "similar names"-issue.

Concerning the "speaking" names, I am a bit reluctant to change them ... - for one, I'm not quite so convinced that shorter names make the code significantly more readable here (the problem is rather the construction as such, which requires mixing and matching these matrix blocks) and secondly, I would prefer to leave them the way they are for consistency: We have used these long, "speaking" names everywhere so it would only make sense to stick to that here. 

Of course, if you feel very strongly about having those variable names changed, I can still do that :)